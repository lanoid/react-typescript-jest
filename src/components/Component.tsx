import * as React from 'react';
import FunctionalComponent from './FunctionalComponent';
import helper from '../helpers/helper';

interface Props {
    hello: string;
};

export default class Component extends React.Component<Props> {

    static defaultProps: Props = {
        hello: 'world',
    }

    render () {
        const { hello } = this.props;
        const greeting = helper();
        
        return (
            <div>
                <h1>{greeting} {hello}</h1>
                <FunctionalComponent />
            </div>
        )
    }
}