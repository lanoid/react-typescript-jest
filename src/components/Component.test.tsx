import * as React from 'react';
import { shallow } from 'enzyme';
import Component from './Component';

describe('Component', () => {
    const year = shallow(<Component />);
    
    it('matches snapshot', () => {
        expect(year).toMatchSnapshot();
    });
});