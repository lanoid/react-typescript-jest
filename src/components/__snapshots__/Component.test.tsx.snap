// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component matches snapshot 1`] = `
<ContextProvider
  value={
    Object {
      "userId": "LAN01D",
    }
  }
>
  <div>
    <h1>
      Hello
       
      world
    </h1>
    <FunctionalComponent
      name="lanoid"
    />
  </div>
</ContextProvider>
`;
